### PR TITLE
SagePayForm: Use valid delimiter in VendorTxCode

### PR DESCRIPTION
**Pull request by aprofeit**
#### Problem
I had a creepy feeling after we merged https://github.com/activemerchant/offsite_payments/commit/e9be51c5cf5be7a1ca8bb7ead24ac68808a0bed5. I checked the [SagePay docs](http://www.sagepay.co.uk/file/25041/download-document/FORM_Integration_and_Protocol_Guidelines_270815.pdf?token=3CMfQadq84V5Z5ktn2xWOTnuQT-OzGtdMG66sSz2GA4) and realized `;` is not a valid character for `VendorTxCode`. Close one.

#### Changes
Use `-` instead.

#### Review
@ThereExistsX @ivanfer

### CODE REVIEW

File: lib/offsite_payments/integrations/sage_pay_form.rb [hunk 1]
```diff
@@ -252,7 +252,7 @@ def message
 
         # Vendor-supplied code (:order mapping).
         def item_id
-          params['VendorTxCode'].split(';').first
+          params['VendorTxCode'].rpartition('-').first
```
Comment (ivanfer) [inline lib/offsite_payments/integrations/sage_pay_form.rb]: what's the difference between this and split?

**Comment by ivanfer** [comment]
nice catch - :pal

File: lib/offsite_payments/integrations/sage_pay_form.rb (hunk 1 above)
Comment (aprofeit) [inline lib/offsite_payments/integrations/sage_pay_form.rb]: If you have a string like `abc-123-def` `split` doesn't have the capability of returning `abc-123`.

```
irb(main):003:0> "abc-123-def".split('-')
=> ["abc", "123", "def"]
irb(main):004:0> "abc-123-def".split('-', 1)
=> ["abc-123-def"]
irb(main):005:0> "abc-123-def".split('-', 2)
=> ["abc", "123-def"]
irb(main):006:0> "abc-123-def".split('-', 3)
=> ["abc", "123", "def"]
irb(main):007:0> "abc-123-def".rpartition('-')
=> ["abc-123", "-", "def"]
irb(main):008:0>
```

**Comment by ThereExistsX** [comment]
I was going to mention that you shouldn't split on `-` for that very reason, good save. You'll need to update the helper tests though.

**Comment by aprofeit** [comment]
In the tests I have `order-28-1234` as the `VendorTxCode` on the test requests just to make sure it splits it correctly. Fixed tests, will be green now.

**Pull request closed without merging**